pyTigerGraph/pyTigerGraph.py

```diff
@@ -960,54 +987,123 @@ def runInterpretedQuery(self, queryText, params=None):
+                    if not graphOnly:  # Vertices and edges are coming in lists; but complex data types too, so more check are needed later
+                        ou.append({o2: o3})
+                else:
+                    ox = o3[0]
+                    if not isinstance(ox, dict):
+                        ou.append({o2: o3})
+                    else:
+                        if "v_type" in ox:
+                            for o4 in o3:
+                                vt = o4["v_type"]
+                                if vt not in vs:
+                                    vs[vt] = []
+                                vs[vt].append(o4)
+                        elif "e_type" in ox:
+                            for o4 in o3:
+                                et = o4["e_type"]
+                                if et not in es:
+                                    es[et] = []
+                                es[et].append(o4)
+                        elif not graphOnly:
+                            ou.append({o3: ox})
+            i += 1
+        ret = {"Vertices": vs, "Edges": es}
+        if not graphOnly:
+            ret["Output"] = ou
+        return ret
 
-    def getVertexDataframeByID(self, vertexType, vertexIds):
-        """Retrieves vertices of the given vertex type, identified by their ID.
+    # Pandas DataFrame support =================================================
 
-        Arguments
-        - `vertexType`: Type of vertex desired
-        - `vertexIds`: A single vertex ID or a list of vertex IDs.
+    def vertexSetToDataFrame(self, vertexSet, withId=True, withType=False):
+        """Converts a vertex set to Pandas DataFrame
 
-        Endpoint:      GET /graph/{graph_name}/vertices
-        Documentation: https://docs.tigergraph.com/dev/restpp-api/built-in-endpoints#get-graph-graph_name-vertices
+        Vertex set is assumed to have this structure:
+        [
+            {
+                "v_id": <vertex_id>,
+                "v_type": <vertex_type_name>,
+                "attributes":
+                    {
+                        "attr1": <value1>,
+                        "attr2": <value2>,
+                         ⋮
+                    }
+            },
+                ⋮
+        ]
         """
-        data = self.getVerticesById(vertexType, vertexIds)
-        df = pd.DataFrame(data)
-        df = pd.concat([df["v_id"], pd.DataFrame(df["attributes"].tolist())], axis=1)
-        return df
-
-    def upsertVertexDataframe(self, df, vertexType, v_id=None, attributes=None):
-        """Upserts vertices from a Pandas data frame.
+        df = pd.DataFrame(vertexSet)
+        cols = []
+        if withId:
+            cols.append(df["v_id"])
+        if withType:
+            cols.append(df["v_type"])
+        cols.append(pd.DataFrame(df["attributes"].tolist()))
+        return pd.concat(cols, axis=1)
+
+    def edgeSetToDataFrame(self, edgeSet, withId=True, withType=False):
```

Comment:
Same here. Please explain that this is the standard way and edge set would come out of a select query.

Reply:
Provided more details/explanation for this function.

Reply:
Nice!